File: projet_stage/car_managment/src/main/java/com/example/car_managment/problem/entities/Problem.java
package com.example.car_managment.problem.entities;

import com.example.car_managment.employees.entites.Employee;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Table(name = "problems")
public class Problem {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  @ManyToOne
  @JoinColumn(name = "employee_id")
  private Employee employee; // Association avec l'entité Employee

  @Column(nullable = false)
  @Enumerated(EnumType.STRING)
  private ProblemStatus status; // Statut du problème

  @Column(nullable = false)
  private String description; // Description du problème

  @Column(nullable = false)
  private LocalDateTime timestamp = LocalDateTime.now(); // Date de déclaration du problème

  public enum ProblemStatus {
    REPORTED,
    IN_PROGRESS,
    REACTED,
    RESOLVED
  }
}

File: projet_stage/car-management-dashboard/src/app/problem.ts
// src/app/services/problem.service.ts

export interface Problem {
    id: number;
    employeeId: number;
    description: ProblemStatus;
    timestamp: Date;
    status: ProblemStatus;
}

export enum ProblemStatus {
    REPORTED = 'REPORTED',
    IN_PROGRESS = 'IN_PROGRESS',
    REACTED = 'REACTED',
    RESOLVED = 'RESOLVED'
}
